ui/apps/platform/src/Containers/Vulnerabilities/WorkloadCves/Tables/SingleEntityVulnerabilitiesTable.tsx

```diff
@@ -1,44 +1,69 @@
 import React from 'react';
 import { Button, ButtonVariant } from '@patternfly/react-core';
-import { TableComposable, Thead, Tr, Th, Tbody, Td } from '@patternfly/react-table';
+import {
+    TableComposable,
+    Thead,
+    Tr,
+    Th,
+    Tbody,
+    Td,
+    ExpandableRowContent,
+} from '@patternfly/react-table';
 import { CheckCircleIcon, ExclamationCircleIcon } from '@patternfly/react-icons';
 import { SVGIconProps } from '@patternfly/react-icons/dist/js/createIcon';
 
 import LinkShim from 'Components/PatternFly/LinkShim';
 import SeverityIcons from 'Components/PatternFly/SeverityIcons';
+import useSet from 'hooks/useSet';
 import { vulnerabilitySeverityLabels } from 'messages/common';
 import { getDistanceStrictAsPhrase } from 'utils/dateUtils';
 import { ImageVulnerabilitiesResponse } from '../hooks/useImageVulnerabilities';
 import { getEntityPagePath } from '../searchUtils';
+import ImageComponentsTable from './ImageComponentsTable';
+import { ImageDetailsResponse } from '../hooks/useImageDetails';
 
 export type SingleEntityVulnerabilitiesTableProps = {
+    image: ImageDetailsResponse['image'] | undefined;
     imageVulnerabilities: ImageVulnerabilitiesResponse['image']['imageVulnerabilities'];
 };
 
 function SingleEntityVulnerabilitiesTable({
+    image,
     imageVulnerabilities,
 }: SingleEntityVulnerabilitiesTableProps) {
+    const expandedRowSet = useSet<string>();
     return (
         <TableComposable>
             <Thead>
                 <Tr>
+                    <Th>{/* Header for expanded column */}</Th>
                     <Th>CVE</Th>
                     <Th>Severity</Th>
-                    <Th>CVE Status</Th>
+                    <Th>CVE status</Th>
                     <Th>Affected components</Th>
                     <Th>First discovered</Th>
                 </Tr>
             </Thead>
-            <Tbody>
-                {imageVulnerabilities.map(
-                    ({ cve, severity, isFixable, imageComponents, discoveredAtImage }) => {
-                        const SeverityIcon: React.FC<SVGIconProps> | undefined =
-                            SeverityIcons[severity];
-                        const severityLabel: string | undefined =
-                            vulnerabilitySeverityLabels[severity];
+            {imageVulnerabilities.map(
+                (
+                    { cve, severity, summary, isFixable, imageComponents, discoveredAtImage },
+                    rowIndex
+                ) => {
+                    const SeverityIcon: React.FC<SVGIconProps> | undefined =
+                        SeverityIcons[severity];
+                    const severityLabel: string | undefined = vulnerabilitySeverityLabels[severity];
```

Comment:
Maybe this is an answer to the question that I asked above about severity keys?

Reply:
I've been using the long form `LOW_VULNERABILITY_SEVERITY` as the canonical severity type since it is what is declared in our cve.proto.ts file and what is returned in the BE queries. The explicit `undefined` checks here are for a few reasons:

1. The GraphQL API declares the return type of `severity` as `string`, even though from what I can tell in the code  it will always be an enum type as declared in the proto. I'm not sure if there is a technical reason the API isn't declaring this as an enum as well, so I'm staying on the safe side assuming it is a string.
2. We have `"noImplicitAny": false` set in our tsconfig, so indexing into the `SeverityIcons` here silently treats `SeverityIcon` as `any`.
3. Even if the types were strictly declared, we could end up with an `UNKNOWN_VULNERABILITY_SEVERITY` here, which would be a type mismatch between the FE and BE declared types.
4. Due to all of the above, even though indexing into `SeverityIcons` should be exhaustive and safe given a "Severity", there is a small chance we could end up with a React component that is `undefined` which would cause a runtime error here.

Reply:
Thank you for explaining. What you say makes sense.

To my original question, the newest query response adds a second convention for severity keys.

String union in frontend types corresponds to enum with string values in backend types.

It seems (to me) as if `enum` in TypeScript corresponded to other languages and, at least for strings, string union has superseded it.